idl/per_partition_rate_limit_info.idl.hh

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2022-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+namespace db {
+
+namespace per_partition_rate_limit {
+
+struct only_account {};
+
+struct account_and_enforce {
+    uint16_t participating_groups;
+    uint16_t replicas_in_group;
+	uint16_t needed_for_cl_in_group;
+};
```

Comment:
Is there some kind of design document that explains what the series is doing? Why do you need to pass this data to all replicas if I see a coordinator code that decides to  rate-limit? Do you decide on replicas as well? Why?

Reply:
> Is there some kind of design document that explains what the series is doing? Why do you need to pass this data to all replicas if I see a coordinator code that decides to rate-limit? Do you decide on replicas as well? Why?

There is a document in the design notes directory, added in the last commit (probably should have placed it in the first commit instead, my bad). It should answer your questions - you can start from the `Implementation` section and read to the end.

Reply:
Moved the commit introducing the document earlier in the series.